### PR TITLE
[4.0] Replace BS alerts with CE alerts

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

This PR replaces all Bootstrap alerts (apart from the installation views) with custom elements (`<joomla-alert>`)

It also addresses the issue: https://github.com/joomla/joomla-cms/issues/18849

### CODE REVIEW

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -1170,8 +1170,8 @@ protected function displayQueries()
 						}
 					}
 
-					$htmlQuery = '<div class="alert alert-danger">' . JText::_('PLG_DEBUG_QUERY_DUPLICATES') . ': ' . implode('&nbsp; ', $dups) . '</div>'
-						. '<pre class="alert hasTooltip" title="' . JHtml::_('tooltipText', 'PLG_DEBUG_QUERY_DUPLICATES_FOUND') . '">' . $text . '</pre>';
+					$htmlQuery = '<joomla-alert type="danger">' . JText::_('PLG_DEBUG_QUERY_DUPLICATES') . ': ' . implode('&nbsp; ', $dups) . '</div>'
+						. '<pre class="alert hasTooltip" title="' . JHtml::_('tooltipText', 'PLG_DEBUG_QUERY_DUPLICATES_FOUND') . '">' . $text . '</joomla-alert>';
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Revert this line and make the change in the previous line.

File: plugins/system/stats/layouts/message.php [hunk 2]
```diff
@@ -19,8 +19,8 @@
  * @var  array                      $statsData     Array containing the data that will be sent to the stats server
  */
 ?>
-<div class="alert alert-info js-pstats-alert" style="display:none;">
-	<button data-dismiss="alert" class="close" type="button">&times;</button>
+
+<joomla-alert type="info" dismiss="true" class="js-pstats-alert" style="display:none;">
```
Comment (Quy) [inline plugins/system/stats/layouts/message.php:23]: Can we replace style attribute with class `d-none`?

Comment (C-Lodder) [reply plugins/system/stats/layouts/message.php:23]: Yup, but that will require JS changes and is out of the scope of this PR. I'll sort that in another PR after this is merged.

File: plugins/system/debug/debug.php (hunk 1 above)
Comment (C-Lodder) [reply plugins/system/debug/debug.php]: @Quy done

**Review by dgrammatiko — APPROVED** [review]

File: administrator/components/com_menus/tmpl/item/edit_container.php [hunk 3]
```diff
@@ -42,7 +42,7 @@
 		<ul class="treeselect">
 			<?php if (count($menuLinks)) : ?>
 				<?php $prevlevel = 0; ?>
-				<div class="alert alert-info"><?php echo JText::_('COM_MENUS_ITEM_FIELD_COMPONENTS_CONTAINER_HIDE_ITEMS_DESC')?></div>
+				<joomla-alert type="info"><?php echo JText::_('COM_MENUS_ITEM_FIELD_COMPONENTS_CONTAINER_HIDE_ITEMS_DESC'); ?></joomla-alert>
```
Comment (Quy) [inline administrator/components/com_menus/tmpl/item/edit_container.php:45]: Fix lines 99-101.

**Comment by Quy** [comment]
Update:
https://github.com/C-Lodder/joomla-cms/blob/9601280e99f413e4d8607e458e553cba1dba70b2/administrator/components/com_modules/tmpl/module/edit_assignment.php#L115

**Comment by C-Lodder** [comment]
@Quy all done

**Comment by Quy** [comment]
There are a few of the following. Should they be updated too?
`<p class="alert alert-info">`

**Comment by C-Lodder** [comment]
ah yes, didn't realise some were initiated with a `<p>` tag. Will do that in a bit

**Comment by Quy** [comment]
More: https://github.com/C-Lodder/joomla-cms/blob/564592a05552c9954d3589df7f828339a4ac9cd2/administrator/components/com_config/tmpl/component/default.php#L67

Also in Atum template files. Here is one.
https://github.com/C-Lodder/joomla-cms/blob/564592a05552c9954d3589df7f828339a4ac9cd2/administrator/templates/atum/error.php#L70

In .xml file, when `alert` is specified in the class attribute:
```
		<field
			name="warning" 
			type="note"
			label="COM_TEMPLATES_CONFIG_SUPPORTED_LABEL"
			default="zip"
			extension="com_templates"
			class="alert alert-info"
		/>
```
here is the output not using CE alert:
```
<div  class="alert alert-info"><h4>Supported File Formats</h4>										</div>
```

**Comment by C-Lodder** [comment]
@Quy - The alerts inside `<noscript>` tags (in the templates) need to stay as they are because they're rendered using Javascript

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 82916aa45de9934efe14e9b3778024708fc46e02<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18872">issues.joomla.org/tracker/joomla-cms/18872</a>.</sub>

**Comment by wilsonge** [comment]
This conflicts with the PR I merged earlier - if you get conflicts sorted will get this merged :)

**Comment by C-Lodder** [comment]
Le fixed

**Comment by wilsonge** [comment]
Thankyou :)

**Pull request merged**